test/browserMonkeySpec.js

```diff
@@ -93,6 +93,21 @@ describe('browser-monkey', function () {
     });
   });
 
+  it('should eventually select an option element', function(){
+    var promise = browser.find('.element').select({text: 'Second'});
+    var selected = false;
+
+    eventuallyInsertHtml(
+      $('<select class="element"><option>First</option><option>Second</option></select>').change(function (e) {
+        selected = true;
+      })
+    );
+
+    return promise.then(function () {
+      expect(selected).to.equal(true);
```

Comment:
Ah!
You are right it only checks that something changed.
I'll fix that

On 3 July 2015 at 08:50, Tim Macfarlane <notifications@github.com> wrote:

> In test/browserMonkeySpec.js
> <https://github.com/featurist/browser-monkey/pull/1#discussion_r33845601>:
>
> > @@ -93,6 +93,21 @@ describe('browser-monkey', function () {
> >      });
> >    });
> >
> > +  it('should eventually select an option element', function(){
> > +    var promise = browser.find('.element').select({text: 'Second'});
> > +    var selected = false;
> > +
> > +    eventuallyInsertHtml(
> > +      $('<select class="element"><option>First</option><option>Second</option></select>').change(function (e) {
> > +        selected = true;
> > +      })
> > +    );
> > +
> > +    return promise.then(function () {
> > +      expect(selected).to.equal(true);
>
> How do you know which one was selected?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/featurist/browser-monkey/pull/1/files#r33845601>.
>